lib/actions/device.coffee

```diff
@@ -208,16 +204,13 @@ exports.await =
 		options.interval ?= 3000
 
 		poll = ->
-			resin.models.device.isOnline params.name, (error, isOnline) ->
-				return done(error) if error?
-
+			resin.models.device.isOnline(params.name).then (isOnline) ->
 				if isOnline
 					console.info("Device became online: #{params.name}")
-					return done()
 				else
 					console.info("Polling device network status: #{params.name}")
 					setTimeout(poll, options.interval)
-
+			.nodeify(done)
```

Comment:
Putting `.nodeify()` here is not correct. The logic of this function is:

- Check if the device is online.
- If it is, exit.
- It it's not, wait for some seconds and go back to step 1.

But as it is now, the function calls `done()` after checking the device network status, without caring about it's results.

What I suggest is:

- Use `.catch(done)` in this line instead of `.nodeify(done)`, so we still react just to errors correctly, and manually call `return done()` if the device is online as it was before.